doc/source/reference/extensions.rst

```diff
@@ -18,6 +18,8 @@ objects.
    api.extensions.register_series_accessor
    api.extensions.register_index_accessor
    api.extensions.ExtensionDtype
+   api.extensions.is_bool_indexer
+   api.extensions.check_bool_array_indexer
```

Comment:
is there a real reason to expose check_book_array_indexer now? 
is_bool_indexer also shouldn't be in api.extension, api.types would be ok, though it was never meant to be public.

can we defer both of these as exposing to the public until we actually see use?

Reply:
Happy to defer exposing those utilities. Will push an update this evening.